src/box/txn.h

```diff
@@ -146,6 +146,38 @@ enum {
 	TXN_SIGNATURE_ABORT = JOURNAL_ENTRY_ERR_MIN - 4,
 };
 
+/** \cond public */
+/**
+ * When a transaction calls `commit`, this action can last for some time until
+ * redo data is written to WAL. While such a `commit` call is in progress we
+ * call changes of such a transaction as 'committed', and when the process is
+ * finished - we call the changes as 'confirmed'. One of the main options of
+ * a transaction is to see or not to see 'committed' changes.
+ * Warning: this enum is exposed in lua via ffi, and thus any change in items
+ * must be correspondingly modified on ffi.cdef(), see schema.lua.
+ */
+enum txn_isolation_level {
+	/** Take isolation level from global default_isolation_level. */
+	TXN_ISOLATION_DEFAULT,
```

Comment:
Why would anyone want to set isolation level to the default value explicitly? If unset, it'll use the default. To reset it after setting it to some level? That would be useless, because the level can't be changed on the flight (this wouldn't make any sense, anyway). I think we should remove the default level from this enum.

Another reason to drop it is that it makes the isolation level difficult to use in the core code internally: instead of simply branching by isolation level, we should also explicitly check the default.

Reply:
Without default isolation we cannot distinguish situations when the level is set or not. That is needed at least in iproto.